### PR TITLE
Allow resizer per format

### PR DESCRIPTION
**Pull request by xaben**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #370 

This PR allows configuring a custom resizer per format, instead of the default resizer configured for the provider.

### CODE REVIEW

**Comment by xaben** [comment]
@rande ping!

**Comment by mvhirsch** [comment]
:+1:

**Comment by bertrandg** [comment]
:+1:

**Comment by Floyd25** [comment]
Will this ever be merged?

File: DependencyInjection/Compiler/AddProviderCompilerPass.php [hunk 1]
```diff
@@ -129,6 +129,7 @@ public function applyFormats(ContainerBuilder $container, array $settings)
                     $config['format']       = isset($config['format'])  ? $config['format'] : 'jpg';
                     $config['height']       = isset($config['height'])  ? $config['height'] : false;
                     $config['constraint']   = isset($config['constraint'])  ? $config['constraint'] : true;
+                    $config['resizer']      = isset($config['resizer'])  ? $config['resizer'] : false;
```
Comment (Soullivaneuh) [inline DependencyInjection/Compiler/AddProviderCompilerPass.php]: Extra space need to be removed:

```php
isset($config['resizer']) ? $config['resizer'] : false;
```

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -81,6 +81,7 @@ private function addContextsSection(ArrayNodeDefinition $node)
                                         ->scalarNode('quality')->defaultValue(80)->end()
                                         ->scalarNode('format')->defaultValue('jpg')->end()
                                         ->scalarNode('constraint')->defaultValue(true)->end()
+                                        ->scalarNode('resizer')->defaultValue(false)->end()
```
Comment (Soullivaneuh) [inline DependencyInjection/Configuration.php]: `->defaultFalse()`

File: Resources/doc/reference/advanced_configuration.rst [hunk 3]
```diff
@@ -26,7 +26,7 @@ Full configuration options:
 
                 formats:
                     small: { width: 100 , quality: 70}
-                    big:   { width: 500 , quality: 70}
+                    big:   { width: 500 , quality: 70, resizer: sonata.media.resizer.square}
```
Comment (Soullivaneuh) [inline Resources/doc/reference/advanced_configuration.rst]: Missing space at end.

File: Resources/doc/reference/media_context.rst [hunk 4]
```diff
@@ -22,7 +25,7 @@ picture format with different sizes and providers. For example:
 
             formats:
                 small: { width: 100 , quality: 70}
-                big:   { width: 500 , quality: 70}
+                big:   { width: 500 , quality: 70, resizer: sonata.media.resizer.square}
```
Comment (Soullivaneuh) [inline Resources/doc/reference/media_context.rst]: Missing space at end.

File: Resources/doc/reference/media_context.rst [hunk 5]
```diff
@@ -31,7 +34,7 @@ picture format with different sizes and providers. For example:
 
             formats:
                 small: { width: 150 , quality: 95}
-                big:   { width: 500 , quality: 90}
+                big:   { width: 500 , quality: 90, resizer: sonata.media.resizer.simple}
```
Comment (Soullivaneuh) [inline Resources/doc/reference/media_context.rst]: Missing space at end.

File: Thumbnail/FormatThumbnail.php [hunk 6]
```diff
@@ -27,6 +30,31 @@ public function __construct($defaultFormat)
...
+
+    /**
+     * @param string $id
+     * @throws \Exception
+     * @return ResizerInterface
+     */
+    public function getResizer($id)
+    {
+        if (!isset($this->resizers[$id])) {
+            throw new \Exception('Resizer with id: '.$id .' is not attached.');
```
Comment (Soullivaneuh) [inline Thumbnail/FormatThumbnail.php]: No space on concatenation operator.

**Comment by Soullivaneuh** [comment]
@xaben I made a quick review.

Ping @rande for the feature.

**Comment by Floyd25** [comment]
Thx for the quick review!

**Comment by xaben** [comment]
@Soullivaneuh Fixed CS as requested.
Ping @rande @Floyd25 .

**Comment by Soullivaneuh** [comment]
@xaben At least, commit have to be squashed before PR got merged (if accepted).

**Comment by xaben** [comment]
@Soullivaneuh  I squashed the commits before pushing the CS fixes. There is only 1 commit at the moment.

**Comment by Soullivaneuh** [comment]
Indeed, didn't notice. Sorry for that.

**Comment by aivus** [comment]
@xaben is it not actual already?

**Comment by xaben** [comment]
@aivus Just waiting for some kind of answer for more then half a year is not normal.

So I closed all my PRs to Sonata project. My time is better spent elsewhere.

**Comment by nicolasricci** [comment]
ping @Soullivaneuh and @rande 
not sure why this has never been merged but it seems a very interesting and currently missing feature of the MediaBundle. 
Should it be reviewed and be reopened?

**Comment by nan-guo** [comment]
why this has never been merged ???

**Comment by OskarStark** [comment]
@nicolasricci @nan-guo 

please feel free to reopen a new PR with the existing code

**Pull request closed without merging**